openwisp_controller/config/base/vpn.py

```diff
@@ -187,6 +200,11 @@ def _validate_subnet_ip(self):
                     {'ip': _('VPN IP address must be within the VPN subnet')}
                 )
 
+    def _validate_authtoken(self):
+        if self._is_backend_type('zerotier'):
+            if not self.auth_token:
+                raise ValidationError({'auth_token': _('Auth token is required')})
+
```

Comment:
@pandafy I tried changing `vpn.js` but couldn't achieve the expected results because the `auth_token` field of the VPN is set to `blank=True`, allowing it to be empty or blank.

```js
// For Zerotier VPN backend
if (backendValue.includes('zerotier')) {
    $('label[for="id_auth_token"]').parent().parent().show();
    // Enhance the user experience by adding the required HTML attribute 
    // to the auth_token field when the zerotier backend is enabled.
    $('label[for="id_auth_token"]').addClass('required')
    $('#id_auth_token').prop('required', true);
} else {
    $('#id_auth_token').prop('required', false);
    $('label[for="id_auth_token"]').removeClass('required')
}
```

Therefore, I overrode the `clean()` method in the `VpnForm` to achieve the desired outcome.

```python
def clean(self):
    cleaned_data = super().clean()
    backend = cleaned_data.get('backend')
    auth_token = cleaned_data.get('auth_token')
    if backend == app_settings.VPN_BACKENDS[3][0] and not auth_token:
        self.add_error(
            'auth_token', 'This field is required for the ZeroTier backend.'
        )

    return cleaned_data

```

Reply:
This is a strange behaviour that it does not respect the `required` attribute.